### PR TITLE
fix: avoid creating blops for UTF-8 text files

### PR DESCRIPTION
**Pull request by rmartine-ias**
To open a pull request using GitHub's API, it's necessary to create a tree, where you can either specify file contents as a string, or the hash of a blob that's already been pushed to GitHub. Creating blobs requires an API call, while specifying file contents does not. GitHub has a low secondary rate limit for creating blobs.  

This PR makes it so fewer files are sent as binary blobs. UTF-8 files do not need to be sent up as blobs, they can be included in the tree as their literal contents.

The fixture update is a bit of a mess but I'm not sure if I can do it better. 

See also: https://github.com/gr2m/octokit-plugin-create-pull-request/issues/121

### CODE REVIEW

File: test/fixtures/happy-path-with-mode.json
Comment (gr2m) [inline test/fixtures/happy-path-with-mode.json:1]: any chance you could just remove the blob creation request instead of updating the whole fixture?

**Comment by gr2m** [comment]
> GitHub has a low secondary rate limit for creating blobs

How do you know? I'm not aware this is the case

**Comment by rmartine-ias** [comment]
> > GitHub has a low secondary rate limit for creating blobs
> 
> How do you know? I'm not aware this is the case

[Experience](https://github.com/backstage/backstage/issues/17188), I looked for docs on the secondary rate limits but didn't find anything concrete. [Others are also unable to find documentation](https://stackoverflow.com/questions/73644164/where-can-i-view-the-secondary-rate-limit-of-github-rest-api). The secondary limit doesn't appear to be returned in headers, either -- tested with an app and user token :/

File: test/fixtures/happy-path-with-mode.json
Comment (rmartine-ias) [reply test/fixtures/happy-path-with-mode.json:1]: done

File: src/value-to-tree-object.ts [hunk 1]
```diff
@@ -8,20 +8,28 @@ export async function valueToTreeObject(
...
-      mode: mode,
+      mode: defaultMode,
       content: value,
     };
   }
 
+  const mode = value.mode ?? defaultMode;
+
+  // UTF-8 files can be treated as text files
+  if (value.encoding === "utf-8") {
```
Comment (gr2m) [inline src/value-to-tree-object.ts:25]: I'd like to add a reference to your pull request as it provides great background information

```suggestion
  // https://github.com/gr2m/octokit-plugin-create-pull-request/pull/133
  if (value.encoding === "utf-8") {
```

**Review by gr2m — APPROVED** [review]
great pull request, thanks a bunch 💐

**Pull request merged**